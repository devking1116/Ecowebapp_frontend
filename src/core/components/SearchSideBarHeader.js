/* @flow */
import React from 'react';
import reactCSS from 'reactcss';
import { Icon } from 'semantic-ui-react';

type Props = {
  title: string,
  catigories?: Array<{
    catigoriesId: string,
    label: string,
  }>,
  isExpanded?: bool,
  expandSideBar: () => void,
  collapseSideBar: () => void,
  children?: any,
};

/**
 * The sidebar header used by a SearchSideBar.
 */
export default function SearchSideBarHeader(props: Props): React.Element<*> {
  const { title, catigories, isExpanded, expandSideBar, collapseSideBar, ...otherProps } = props;

  const styles = reactCSS({
    'default': {
      wrapper: {
        display: 'flex',
        justifyContent: 'flex-end',
        height: 20,
      },
      title: {
        color: '#5e656a',
      },
    },
    'isExpanded': {
      wrapper: {
        justifyContent: 'space-between',
      },
    },
  }, { isExpanded });

  if(!isExpanded) {
    return (
      <div style={styles.wrapper} {...otherProps}>
        <Icon name="content" size="large" onClick={expandSideBar} />
      </div>
    );
  }

  return (
    <div style={styles.wrapper} {...otherProps}>
      <h3 style={styles.title}>
        {title}
      </h3>
    {catigories?
      <div>NYI: catigorie selector</div>:
      null
    }
      <Icon name="minus circle" size="large" color="grey" inverted onClick={collapseSideBar} />
    </div>
  );
}
