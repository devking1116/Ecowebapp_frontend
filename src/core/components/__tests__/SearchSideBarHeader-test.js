import React from 'react';
import renderer from 'react-test-renderer';

import SearchSideBarHeader from '../SearchSideBarHeader';

test('the open button is displayed, when collapsed', () => {
  const component = renderer.create(
    <SearchSideBarHeader title="Test Title" />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('the title is displayed, when expaned', () => {
  const component = renderer.create(
    <SearchSideBarHeader title="Test Title" isExpanded />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('the catigories selector is displayed, if catigories are given', () => {
  const component = renderer.create(
    <SearchSideBarHeader
      catigories={[
        { catigoriesId: 0, label: 'Test catigory 0'},
        { catigoriesId: 1, label: 'Test catigory 1'},
        { catigoriesId: 2, label: 'Test catigory 2'},
      ]}
      isExpanded
    />
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

test('when close button is clicked, if expanded, it calls the collapse function.', () => {
  const collapseFn = jest.fn();
  const component = renderer.create(
    <SearchSideBarHeader collapseSideBar={collapseFn} isExpanded />
  );
  const tree = component.toJSON();
  const [ _, icon ] = tree.children;
  const { onClick } = icon.props;

  onClick();
  expect(collapseFn).toHaveBeenCalledTimes(1);
});

test('when open icon is clicked, if collapsed, it calls the expand function.', () => {
  const expandFn = jest.fn();
  const component = renderer.create(
    <SearchSideBarHeader expandSideBar={expandFn} />
  );
  const tree = component.toJSON();
  const [ icon ] = tree.children;
  const { onClick } = icon.props;

  onClick();
  expect(expandFn).toHaveBeenCalledTimes(1);
});
