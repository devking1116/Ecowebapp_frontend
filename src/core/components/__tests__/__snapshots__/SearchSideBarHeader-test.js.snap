exports[`test the catigories selector is displayed, if catigories are given 1`] = `
<div
  style={
    Object {
      "WebkitJustifyContent": "space-between",
      "display": "flex",
      "height": 20,
      "justifyContent": "space-between",
    }
  }>
  <h3
    style={
      Object {
        "color": "#5e656a",
      }
    } />
  <div>
    NYI: catigorie selector
  </div>
  <i
    aria-hidden="true"
    className="grey minus circle large inverted icon"
    onClick={undefined} />
</div>
`;

exports[`test the open button is displayed, when collapsed 1`] = `
<div
  style={
    Object {
      "WebkitJustifyContent": "flex-end",
      "display": "flex",
      "height": 20,
      "justifyContent": "flex-end",
    }
  }>
  <i
    aria-hidden="true"
    className="content large icon"
    onClick={undefined} />
</div>
`;

exports[`test the title is displayed, when expaned 1`] = `
<div
  style={
    Object {
      "WebkitJustifyContent": "space-between",
      "display": "flex",
      "height": 20,
      "justifyContent": "space-between",
    }
  }>
  <h3
    style={
      Object {
        "color": "#5e656a",
      }
    }>
    Test Title
  </h3>
  <i
    aria-hidden="true"
    className="grey minus circle large inverted icon"
    onClick={undefined} />
</div>
`;
